---
title: SPEAQeasy bootcamp
author: 
  - name: Leonardo Collado-Torres
    affiliation:
    - &libd Lieber Institute for Brain Development, Johns Hopkins Medical Campus
    - &ccb Center for Computational Biology, Johns Hopkins University
    email: lcolladotor@gmail.com
output: 
  BiocStyle::html_document:
    toc: true
    toc_float: true
    toc_depth: 2
    code_folding: show
---

This document is part of the _R/Bioconductor-powered Team Data Science_ [LIBD bootcamps](https://lcolladotor.github.io/bioc_team_ds/rbioconductor-data-science-bootcamps.html#libd-bootcamps).

You can download the [**latest version of this Rmd document here**](https://github.com/LieberInstitute/SPEAQeasy-example/blob/master/bootcamp_intro.Rmd). ^[This was useful particularly during October 5-7 while I was updating this document frequently.]

# Setup

Please install the latest:

* R version from [CRAN](https://cran.r-project.org/)
* [RStudio Destkop](https://rstudio.com/products/rstudio/download/) ^[You might want to try out [RStudio Desktop Preview](https://rstudio.com/products/rstudio/download/preview/) as discussed on the [2020-10-02 LIBD rstats club session](https://twitter.com/LIBDrstats/status/1312187753371561984?s=20). ]

Once you have installed R and RStudio, please install the following R packages:

```{r "install_packages", eval = FALSE}
## For installing R packages from CRAN and GitHub
if (!requireNamespace("remotes", quietly = TRUE)) {
    install.packages("remotes")
}

## For installing Bioconductor packages
remotes::install_cran("BiocManager")

## Workflow-related R packages
remotes::install_cran(c(
    "rmarkdown",
    "sessioninfo",
    "getopt",
    "here",
    "tidyr",
    "usethis"
))
remotes::install_github("LieberInstitute/jaffelab")

## Differential-expression related R packages
BiocManager::install(c("limma", "edgeR", "SummarizedExperiment", "recount"))

## Genomics-related R packages
BiocManager::install(c("clusterProfiler", "org.Hs.eg.db"))

## Visualization-related R packages
remotes::install_cran(c(
    "pheatmap",
    "RColorBrewer",
    "VariantAnnotation",
    "ggplot2",
    "plotly"
))
BiocManager::install("iSEE")

## Extra ones
remotes::install_cran(c("knitcitations", "statmod"))
BiocManager::install(c("BiocStyle"))
```

# Bootcamp plan

* Day 1: Monday October 5th 2020, 3-5 pm
  - RNA sequencing primer
  - LIBD's RNA-seq processing pipeline: SPEAQeasy
  - `r BiocStyle::Biocpkg("SummarizedExperiment")` overview
  - Exploring quality control metrics
* Day 2: Tuesday October 6th 2020, 3-5 pm
  - Exploring gene expression
  - Identifying sample swaps
  - Statistical modeling
* Day 3: Wednesday October 7th 2020, 1-3 pm
  - Visualizing differentially expressed genes
  - Gene ontology enrichment analyses
  - Beyond SPEAQeasy
  

# Day 1

## RNA sequencing primer

[Slides](https://docs.google.com/presentation/d/1X3U2lcf75Yh9BK_Scs-O3M8b9-JauUNN8ZO_xhlKs7w/edit?usp=sharing)

For more details, I have some older slides on the technology for RNA-seq from [PDCB-HTS 2010](http://lcolladotor.github.io/courses/PDCB-HTS.html).

## SPEAQeasy overview

Let's go to the [SPEAQeasy](http://research.libd.org/SPEAQeasy/) documentation website.

We'll mostly be using the [SPEAQeasy main output files](http://research.libd.org/SPEAQeasy/outputs.html). 

### SPEAQeasy-example

Example data using [SPEAQeasy-example](http://research.libd.org/SPEAQeasy-example/) with 40 samples from the Bipolar project with Peter Zandi et al.

It shows:

1. How to download the example data
1. How to run SPEAQeasy with this data
1. How to identify sample swaps
1. How to run differential expression analyses with this data

We'll view these links, but we'll start with:

* [Home](http://research.libd.org/SPEAQeasy-example/index.html)
* [Download data](http://research.libd.org/SPEAQeasy-example/prepare_data.html)

## SummarizedExperiment overview

### General

In general:

* `r BiocStyle::Biocpkg("SummarizedExperiment")`
  - [Introduction vignette](http://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html)
* Bioconductor 2015 paper
  - DOI: [10.1038/nmeth.3252](https://doi.org/10.1038/nmeth.3252)
  
### Example data

Let's look at the example data

```{r "load_data"}
## For loading the data
library("SummarizedExperiment")
library("here")

## Load the example data
load(here("rse_speaqeasy.RData"), verbose = TRUE)
```

Let's look at the sample metadata we have in this example data. Some of these variables are defined in the [SPEAQeasy documentation](http://research.libd.org/SPEAQeasy/outputs.html#quality-metrics) that we'll be using shortly.

```{r "metadata"}
colData(rse_gene)
```

SPEAQeasy can align data to the human or mouse genomes using specific Gencode annotations. The example data we are working with is from human using Gencode v32.

```{r "gene data"}
rowData(rse_gene)
```

The bulk of the data is stored in the gene count matrix generated using `featureCounts` by `SPEAQeasy`. This is a large matrix with one row per gene and one column per sample. As such, we'll just explore a tiny piece of it.

```{r "count data"}
## The names of the count matrices we have
assayNames(rse_gene)

## The full "counts" matrix
dim(assay(rse_gene, "counts"))

## The top left corner of the matrix counts
jaffelab::corner(assay(rse_gene, "counts"))
```


## Exploring quality control metrics

The [SPEAQeasy documentation](http://research.libd.org/SPEAQeasy/outputs.html#quality-metrics) explains the meaning of some of the quality metrics that are included in the RSE files generated by this processing pipeline. Let's explore these metrics and their relationship to some of our sample metadata we manually included, such as the age at time of death and primary diagnosis.

We can start with some boxplots comparing some metrics against primary diagnosis and race, as shown in the [SPEAQeasy-example differential expression script](http://research.libd.org/SPEAQeasy-example/de_analysis_speaqeasy.html#14_Stats_vs_race_and_brain_region).

We can also use tools like `r BiocStyle::CRANpkg("ggplot2")` and `r BiocStyle::CRANpkg("plotly")` to interactively explore some metrics. For doing so, it'll be useful to check some of our LIBD rstats club sessions on:

* [High quality graphics in R](https://twitter.com/LIBDrstats/status/1308934265267195905?s=20)
* [Interactive graphics with plotly](https://twitter.com/LIBDrstats/status/1304799631386308608?s=20)


### Exercise

Adapt the interactive graphics code from the LIBD rstats club session available [here](https://gist.github.com/lcolladotor/350ac8843e153d135f64880a1d029b14) for our data.


### Exercise hints

```{r "interactive_exploration"}
## First we load the required packages
library("ggplot2")
library("plotly")

## Next, we convert the sample metadata to a data.frame
## which will make it easier to work with
pd_df <- as.data.frame(colData(rse_gene))

## We need a "key" variable that is unique

## Let's make the key

## Let's make a "highlighted" table

## Make a plot using the highlight table

## Make a second plot

## Convert them to interactive plots

## Now group them together

## Fancy one
```

### Exercise solution

#### Step 1

As a first step, let's add the code from the gist that is listed below each of the comments from the previous [code](https://gist.github.com/lcolladotor/350ac8843e153d135f64880a1d029b14).

```{r "interactive_exploration_step1", eval = FALSE}
## First we load the required packages
library("ggplot2")
library("plotly")

## Next, we convert the sample metadata to a data.frame
## which will make it easier to work with
pd_df <- as.data.frame(colData(rse_gene))

## We need a "key" variable that is unique
length(unique(pd_df$RNum))
nrow(pd_df)

## Let's make the key
pd_df$key <- pd_df$RNum

## Let's make a "highlighted" table
pd_key <- highlight_key(pd_df, ~key)

## Make a plot using the highlight table
gg_mean_mito_vs_mean_gene <-
    ggplot(
        pd_key,
        aes(x = mean_mitoRate, y = mean_totalAssignedGene, color = Region)
    ) +
    geom_point()
gg_mean_mito_vs_mean_gene

## Make a second plot
gg_mean_mito_vs_mean_RIN <-
    ggplot(pd_key, aes(x = mean_mitoRate, y = mean_RIN, color = Region)) +
    geom_point()
gg_mean_mito_vs_mean_RIN

## Convert them to interactive plots
p_mean_mito_vs_mean_gene <- ggplotly(gg_mean_mito_vs_mean_gene)
p_mean_mito_vs_mean_RIN <- ggplotly(gg_mean_mito_vs_mean_RIN)

## Now group them together
p_merged <- subplot(
    p_mean_mito_vs_mean_gene,
    p_mean_mito_vs_mean_RIN,
    nrows = 1,
    shareX = TRUE,
    shareY = FALSE,
    which_layout = 2
)

## Fancy one
p_fancy <- highlight(
    p_merged,
    on = "plotly_click",
    selectize = TRUE,
    dynamic = TRUE,
    persistent = TRUE
)
```

#### Step 2

Now that we have the code, we need to adapt it. Some variable names are slightly different, some don't exist. This process will typically involve several error messages while we figure out what's there and what's different. The end result will look like the following code.


```{r "interactive_exploration_answer"}
## First we load the required packages
library("ggplot2")
library("plotly")

## Next, we convert the sample metadata to a data.frame
## which will make it easier to work with
pd_df <- as.data.frame(colData(rse_gene))

## We need a "key" variable that is unique
## We replaced RNum by SAMPLE_ID
length(unique(pd_df$SAMPLE_ID))
nrow(pd_df)

## Let's make the key
pd_df$key <- pd_df$SAMPLE_ID

## Let's make a "highlighted" table
pd_key <- highlight_key(pd_df, ~key)

## Make a plot using the highlight table
## Here we remoted "mean_" from the variable and object names
## and we changed Region to BrainRegion
gg_mito_vs_gene <-
    ggplot(
        pd_key,
        aes(x = mitoRate, y = totalAssignedGene, color = BrainRegion)
    ) +
    geom_point()

## Make a second plot
## We don't have RIN, so we'll use rRNA_rate here
gg_mito_vs_rRNA_rate <-
    ggplot(pd_key, aes(x = mitoRate, y = rRNA_rate, color = BrainRegion)) +
    geom_point()

## Convert them to interactive plots
p_mito_vs_gene <- ggplotly(gg_mito_vs_gene)
p_mito_vs_rRNA_rate <- ggplotly(gg_mito_vs_rRNA_rate)

## Now group them together
p_merged <- subplot(
    p_mito_vs_gene,
    p_mito_vs_rRNA_rate,
    nrows = 1,
    shareX = TRUE,
    shareY = FALSE,
    which_layout = 2
)

## Fancy one
p_fancy <- highlight(
    p_merged,
    on = "plotly_click",
    selectize = TRUE,
    dynamic = TRUE,
    persistent = TRUE
)

## End result
p_fancy
```


# Day 2

You can download the [**latest version of this Rmd document here**](https://github.com/LieberInstitute/SPEAQeasy-example/blob/master/bootcamp_intro.Rmd).

## Exploring gene expression

## Identifying sample swaps

## Statistical modeling

# Day 3

You can download the [**latest version of this Rmd document here**](https://github.com/LieberInstitute/SPEAQeasy-example/blob/master/bootcamp_intro.Rmd).

## Visualizing DEGs

## GO enrichment analyses

## Beyond SPEAQeasy


# Reproducibility

```{r reproducibility}
# Date this report was generated
message(Sys.time())

# Reproducibility info
options(width = 120)
sessioninfo::session_info()
```
