<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2020-07-10" />

<title>SPEAQeasy Differential Expression Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">

</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="/usr/local/lib/R/host-site-library/BiocStyle/resources/html/bioconductor.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 828px;
  margin-left: auto;
  margin-right: auto;
}

img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




<script>
function toggle_visibility(id1) {
  var e = document.getElementById(id1);
  e.style.display = ((e.style.display!="none") ? "none" : "block");
}
</script>

</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SPEAQeasy example</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="pull_data/prepare_data.html">Downloading the example data</a>
</li>
<li>
  <a href="run_speaqeasy.html">Run SPEAQeasy on your setup]</a>
</li>
<li>
  <a href="swap_speaqeasy.html">Resolve identity issues and attach phenotype data</a>
</li>
<li>
  <a href="DE_analysis/de_analysis_speaqeasy.html">Perform differential expression and visualize results</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lcolladotor/twitter-stats">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">SPEAQeasy Differential Expression Analysis</h1>
<h4 class="date">July 10, 2020</h4>

</div>

<h1>Contents</h1>
<div id="TOC">
<ul>
<li><a href="#analysis"><span class="toc-section-number">1</span> Analysis</a><ul>
<li><a href="#load-required-libraries"><span class="toc-section-number">1.1</span> Load required libraries</a></li>
<li><a href="#load-data-and-prepare-directories-to-place-outputs-in"><span class="toc-section-number">1.2</span> Load data and prepare directories to place outputs in</a></li>
<li><a href="#cell-pcs"><span class="toc-section-number">1.3</span> cell PCs</a></li>
<li><a href="#filter-for-expressed"><span class="toc-section-number">1.4</span> filter for expressed</a></li>
<li><a href="#metrics"><span class="toc-section-number">1.5</span> metrics</a></li>
<li><a href="#check-if-ratios-of-cell-changed-by-batch"><span class="toc-section-number">1.6</span> check if ratios of cell changed by batch</a></li>
<li><a href="#pc1-vs-pc2"><span class="toc-section-number">1.7</span> pc1 vs pc2</a></li>
<li><a href="#by-line"><span class="toc-section-number">1.8</span> by line</a></li>
<li><a href="#by-experiment"><span class="toc-section-number">1.9</span> by experiment</a></li>
<li><a href="#modeling"><span class="toc-section-number">1.10</span> modeling</a></li>
<li><a href="#mean-variance"><span class="toc-section-number">1.11</span> mean-variance</a></li>
<li><a href="#check-plots"><span class="toc-section-number">1.12</span> check plots</a><ul>
<li><a href="#make-boxplots"><span class="toc-section-number">1.12.1</span> make boxplots</a></li>
</ul></li>
</ul></li>
<li><a href="#cleangeneexprs-cleaningygeneexprs-modis.naebgenep.value1-p3"><span class="toc-section-number">2</span> cleanGeneExprs = cleaningY(geneExprs, mod[,!is.na(eBGene$p.value[1,])], P=3)</a><ul>
<li><a href="#gene-ontology"><span class="toc-section-number">2.0.1</span> gene ontology</a></li>
<li><a href="#get-significant-genes-by-sign"><span class="toc-section-number">2.1</span> get significant genes by sign</a></li>
<li><a href="#do-go-and-kegg"><span class="toc-section-number">2.2</span> do GO and KEGG</a></li>
</ul></li>
</ul>
</div>

<div id="analysis" class="section level1">
<h1><span class="header-section-number">1</span> Analysis</h1>
<p>The following analysis explores a <code>RangedSummarizedExperiment</code> object from the SPEAQeasy pipeline. Note that we will use a modified version of the object, which resolved sample identity issues which were present in the raw output from SPEAQeasy. This object also includes phenotype data added after resolving identity issues. Though SPEAQeasy produces objects for several feature types (genes, exons, exon-exon junctions), we will demonstrate an example analysis for just genes. We will perform differential expression across some typical variables of interest (e.g. sex, age, race) and show how to perform principal component analysis (PCA) and visualize findings with plots.</p>
<div id="load-required-libraries" class="section level2">
<h2><span class="header-section-number">1.1</span> Load required libraries</h2>
<pre class="r"><code>library(&quot;SummarizedExperiment&quot;)
library(&quot;recount&quot;)
library(&quot;edgeR&quot;)
library(&quot;limma&quot;)
library(&quot;jaffelab&quot;) # GitHub: LieberInstitute/jaffelab
library(&quot;RColorBrewer&quot;)
library(&quot;clusterProfiler&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(&quot;pheatmap&quot;)
library(&quot;here&quot;)</code></pre>
</div>
<div id="load-data-and-prepare-directories-to-place-outputs-in" class="section level2">
<h2><span class="header-section-number">1.2</span> Load data and prepare directories to place outputs in</h2>
<p>For those who ran SPEAQeasy from the example FASTQ data set, the <code>RangedSummarizedExperiment</code> will have a different path, as specified with the <code>--output</code> flag. Set the working directory according to where the <code>SPEAQeasy-example</code> repository is cloned.</p>
<pre class="r"><code># Load the RSE gene object
load(here(&quot;rse_speaqeasy.RData&quot;), verbose = TRUE)</code></pre>
<pre><code>## Loading objects:
##   rse_gene</code></pre>
<pre class="r"><code># Create directories to organize outputs from this analysis
dir.create(here(&quot;DE_analysis&quot;, &quot;pdfs&quot;), showWarnings = FALSE)
dir.create(here(&quot;DE_analysis&quot;, &quot;tables&quot;), showWarnings = FALSE)
dir.create(here(&quot;DE_analysis&quot;, &quot;rdas&quot;), showWarnings = FALSE)</code></pre>
</div>
<div id="cell-pcs" class="section level2">
<h2><span class="header-section-number">1.3</span> cell PCs</h2>
<p>col_names = c(‘trimmed’, ‘numReads’, ‘numMapped’, ‘numUnmapped’,
‘overallMapRate’, ‘concordMapRate’, ‘totalMapped’, ‘mitoMapped’,
‘mitoRate’, ‘totalAssignedGene’)
cellPca = prcomp(as.data.frame(colData(rse_gene)[,col_names]))
rse_gene<span class="math inline">\(PC = cellPca\)</span>x[,1]
getPcaVars(cellPca)[1] # 87.3
round(cellPca$rot[,1],3)
# fetal quiescent and adult neuron increase</p>
</div>
<div id="filter-for-expressed" class="section level2">
<h2><span class="header-section-number">1.4</span> filter for expressed</h2>
<p>rse_gene = rse_gene[rowMeans(getRPKM(rse_gene,“Length”)) &gt; 0.2,]</p>
<div id="section" class="section level14">
<p class="heading"><span class="header-section-number">1.4.0.0.0.0.0.0.0.0.0.0.0.1</span> </p>
</div>
</div>
<div id="metrics" class="section level2">
<h2><span class="header-section-number">1.5</span> metrics</h2>
</div>
<div id="check-if-ratios-of-cell-changed-by-batch" class="section level2">
<h2><span class="header-section-number">1.6</span> check if ratios of cell changed by batch</h2>
<p>pdf(file = here(“DE_analysis”, “pdfs”, “Region_Race_cellcheck.pdf”))
boxplot(rse_gene<span class="math inline">\(rRNA_rate ~ rse_gene\)</span>BrainRegion,xlab="“)
boxplot(rse_gene<span class="math inline">\(mitoRate ~ rse_gene\)</span>BrainRegion,xlab=”“)
boxplot(rse_gene<span class="math inline">\(gene_Assigned ~ rse_gene\)</span>BrainRegion,xlab=”“)
boxplot(rse_gene<span class="math inline">\(mitoRate ~ rse_gene\)</span>Race,las=3,xlab=”“)
boxplot(rse_gene<span class="math inline">\(gene_Assigned ~ rse_gene\)</span>Race,las=3,xlab=”")
dev.off()</p>
<div id="explore-human" class="section level4">
<h4><span class="header-section-number">1.6.0.1</span> explore human</h4>
<p>geneExprs = log2(getRPKM(rse_gene,“Length”)+1)
pca = prcomp(t(geneExprs))
pca_vars = getPcaVars(pca)
pca_vars_lab = paste0(“PC”, seq(along=pca_vars), “:”,
pca_vars, “% Var Expl”)</p>
<div id="section-1" class="section level10">
<p class="heading"><span class="header-section-number">1.6.0.1.0.0.0.0.0.1</span> </p>
<p>pdf(here(“DE_analysis”, “pdfs”, “PCA_plotsExprs.pdf”), w=9)
par(mar=c(8,6,2,2),cex.axis=1.8,cex.lab=1.8)
palette(brewer.pal(4,“Dark2”))</p>
</div>
</div>
</div>
<div id="pc1-vs-pc2" class="section level2">
<h2><span class="header-section-number">1.7</span> pc1 vs pc2</h2>
<p>plot(pca<span class="math inline">\(x, pch=21, bg=factor(rse_gene\)</span>PrimaryDx),cex=1.2,
xlab = pca_vars_lab[1], ylab = pca_vars_lab[2])
legend(“bottomleft”, levels(rse_gene$PrimaryDx), col=1:2, pch=15,cex=2)</p>
</div>
<div id="by-line" class="section level2">
<h2><span class="header-section-number">1.8</span> by line</h2>
<p>for(i in 1:10) {
boxplot(pca<span class="math inline">\(x[,i] ~ rse_gene\)</span>Sex,
ylab=pca_vars_lab[i], las = 3,xlab=“Sex”,outline=FALSE)
points( pca<span class="math inline">\(x[,i] ~ jitter(as.numeric(factor(rse_gene\)</span>Sex))),
pch = 21, bg = rse_gene$PrimaryDx,cex=1.2)
}</p>
</div>
<div id="by-experiment" class="section level2">
<h2><span class="header-section-number">1.9</span> by experiment</h2>
<p>for(i in 1:10) {
boxplot(pca<span class="math inline">\(x[,i] ~ rse_gene\)</span>Race,
ylab=pca_vars_lab[i], las = 3,xlab=“Race”,outline=FALSE)
points( pca<span class="math inline">\(x[,i] ~ jitter(as.numeric(factor(rse_gene\)</span>Race))),
pch = 21, bg = rse_gene$PrimaryDx,cex=1.2)
}
dev.off()</p>
<div id="section-2" class="section level20">
<p class="heading"><span class="header-section-number">1.9.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1</span> </p>
</div>
</div>
<div id="modeling" class="section level2">
<h2><span class="header-section-number">1.10</span> modeling</h2>
<div id="section-3" class="section level20">
<p class="heading"><span class="header-section-number">1.10.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1</span> </p>
<p>dge = DGEList(counts = assays(rse_gene)$counts,
genes = rowData(rse_gene))
dge = calcNormFactors(dge)</p>
</div>
</div>
<div id="mean-variance" class="section level2">
<h2><span class="header-section-number">1.11</span> mean-variance</h2>
<p>mod = model.matrix(~PrimaryDx + PC + BrainRegion,
data=colData(rse_gene))
pdf(file = “DE_analysis/pdfs/vGene.pdf”)
vGene = voom(dge,mod,plot=TRUE)
dev.off()</p>
<p>gene_dupCorr = duplicateCorrelation(vGene<span class="math inline">\(E, mod,  block=colData(rse_gene)\)</span>SAMPLE_ID)
save(gene_dupCorr, file = “DE_analysis/rdas/gene_dupCorr_neurons.rda”)</p>
<p>fitGeneDupl = lmFit(vGene,
correlation=gene_dupCorr<span class="math inline">\(consensus.correlation,  block=colData(rse_gene)\)</span>SAMPLEID)</p>
<p>ebGeneDupl = eBayes(fitGeneDupl)
outGeneDupl = topTable(ebGeneDupl,coef=2,
p.value = 1,number=nrow(rse_gene),sort=“none”)</p>
<p>pdf(file = here(“DE_analysis”, “pdfs”, “hist_pval.pdf”))
hist(outGeneDupl<span class="math inline">\(P.Value) dev.off() table(outGeneDupl\)</span>adj.P.Val &lt; 0.05)
table(outGeneDupl$adj.P.Val &lt; 0.1)</p>
<p>sigGeneDupl = topTable(ebGeneDupl,coef=2,
p.value = 0.1,number=nrow(rse_gene))</p>
<p>sigGeneDupl[,c(“Symbol”,“logFC”, “P.Value”,“AveExpr”)]
sigGeneDupl[sigGeneDupl$logFC &gt; 0,c(“Symbol”,“logFC”, “P.Value”)]
sigGeneDupl[sigGeneDupl$logFC &lt; 0,c(“Symbol”,“logFC”, “P.Value”)]</p>
<p>write.csv(outGeneDupl, file = here(“DE_analysis”, “tables”, “de_stats_allExprs.csv”))
write.csv(sigGeneDupl, file = here(“DE_analysis”, “tables”, “de_stats_fdr10_sorted.csv”))</p>
<div id="section-4" class="section level19">
<p class="heading"><span class="header-section-number">1.11.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1</span> </p>
</div>
</div>
<div id="check-plots" class="section level2">
<h2><span class="header-section-number">1.12</span> check plots</h2>
<div id="section-5" class="section level19">
<p class="heading"><span class="header-section-number">1.12.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1</span> </p>
<p>exprs = vGene$E[rownames(sigGeneDupl),]
#exprsClean = cleaningY(exprs, mod, 2)</p>
</div>
<div id="make-boxplots" class="section level3">
<h3><span class="header-section-number">1.12.1</span> make boxplots</h3>
</div>
</div>
</div>
<div id="cleangeneexprs-cleaningygeneexprs-modis.naebgenep.value1-p3" class="section level1">
<h1><span class="header-section-number">2</span> cleanGeneExprs = cleaningY(geneExprs, mod[,!is.na(eBGene$p.value[1,])], P=3)</h1>
<p>pdf(here(“DE_analysis”, “pdfs”, “DE_boxplots_byDiagnosis.pdf”, w=10))
par(mar=c(8,6,4,2),cex.axis=1.8,cex.lab=1.8, cex.main=1.8)
palette(brewer.pal(4,“Dark2”))
for(i in 1:nrow(sigGeneDupl)) {
yy = exprs[i,]
boxplot(yy ~ rse_gene<span class="math inline">\(PrimaryDx, outline=FALSE,  ylim=range(yy), ylab=&quot;Normalized log2 Exprs&quot;, xlab=&quot;&quot;,  main = paste(sigGeneDupl\)</span>Symbol[i], “-”, sigGeneDupl<span class="math inline">\(gencodeID[i]))  points(yy ~ jitter(as.numeric(rse_gene\)</span>PrimaryDx)),
pch = 21, bg= rse_gene<span class="math inline">\(PrimaryDx,cex=1.3)  ll = ifelse(sigGeneDupl\)</span>logFC[i] &gt; 0, “topleft”, “topright”)
legend(ll, paste0(“p=”, signif(sigGeneDupl$P.Value[i],3)), cex=1.3)
}
dev.off()</p>
<div id="rpkm" class="section level4">
<h4><span class="header-section-number">2.0.0.1</span> RPKM</h4>
<p>e = geneExprs[rownames(sigGeneDupl),]</p>
<p>pdf(here(“DE_analysis”, “pdfs”, “DE_boxplots_byGenome_log2RPKM.pdf”, w=10))
par(mar=c(8,6,4,2),cex.axis=1.8,cex.lab=1.8, cex.main=1.8)
palette(brewer.pal(4,“Dark2”))
for(i in 1:nrow(sigGeneDupl)) {
yy = e[i,]
boxplot(yy ~ rse_gene<span class="math inline">\(PrimaryDx, las=3,outline=FALSE,  ylim=range(yy), ylab=&quot;log2(RPKM+1)&quot;, xlab=&quot;&quot;,  main = paste(sigGeneDupl\)</span>Symbol[i], “-”, sigGeneDupl<span class="math inline">\(gencodeID[i]))  points(yy ~ jitter(as.numeric(rse_gene\)</span>PrimaryDx)),
pch = 21, bg= rse_gene<span class="math inline">\(PrimaryDx,cex=1.3)  ll = ifelse(sigGeneDupl\)</span>logFC[i] &gt; 0, “topleft”, “topright”)
legend(ll, paste0(“p=”, signif(sigGeneDupl$P.Value[i],3)), cex=1.3)
}
dev.off()</p>
<div id="no-rat-astrocyte-differences" class="section level20">
<p class="heading"><span class="header-section-number">2.0.0.1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1</span> no rat astrocyte differences</p>
</div>
</div>
<div id="gene-ontology" class="section level3">
<h3><span class="header-section-number">2.0.1</span> gene ontology</h3>
</div>
<div id="get-significant-genes-by-sign" class="section level2">
<h2><span class="header-section-number">2.1</span> get significant genes by sign</h2>
<p>sigGene = outGeneDupl[outGeneDupl$P.Value &lt; 0.005,]
sigGeneList = split(as.character(sigGene<span class="math inline">\(EntrezID), sign(sigGene\)</span>logFC))
sigGeneList = lapply(sigGeneList, function(x) x[!is.na(x)])
geneUniverse = as.character(outGeneDupl$EntrezID)
geneUniverse = geneUniverse[!is.na(geneUniverse)]</p>
</div>
<div id="do-go-and-kegg" class="section level2">
<h2><span class="header-section-number">2.2</span> do GO and KEGG</h2>
<p>goBP_Adj &lt;- compareCluster(sigGeneList, fun = “enrichGO”,
universe = geneUniverse, OrgDb = org.Hs.eg.db,
ont = “BP”, pAdjustMethod = “BH”, pvalueCutoff = 1,
qvalueCutoff = 1, readable= TRUE)</p>
<p>goMF_Adj &lt;- compareCluster(sigGeneList, fun = “enrichGO”,
universe = geneUniverse, OrgDb = org.Hs.eg.db,
ont = “MF”, pAdjustMethod = “BH”, pvalueCutoff = 1,
qvalueCutoff = 1, readable= TRUE)</p>
<p>goCC_Adj &lt;- compareCluster(sigGeneList, fun = “enrichGO”,
universe = geneUniverse, OrgDb = org.Hs.eg.db,
ont = “CC”, pAdjustMethod = “BH”, pvalueCutoff = 1,
qvalueCutoff = 1, readable= TRUE)</p>
<p>kegg_Adj &lt;- compareCluster(sigGeneList, fun = “enrichKEGG”,
universe = geneUniverse, pAdjustMethod = “BH”,
pvalueCutoff = 1, qvalueCutoff = 1)</p>
<p>save(goBP_Adj, goCC_Adj, goMF_Adj, kegg_Adj,
file = here(“DE_analysis”, “rdas”, “gene_set_objects_p005.rda”))</p>
<p>goList = list(BP = goBP_Adj, MF = goMF_Adj, CC = goCC_Adj, KEGG = kegg_Adj)
goDf = dplyr::bind_rows(lapply(goList, as.data.frame), .id = “Ontology”)
goDf = goDf[order(goDf$pvalue),]</p>
<p>write.csv(goDf, file = here(“DE_analysis”, “tables”, “geneSet_output.csv”), row.names=FALSE)</p>
<p>options(width=130)
goDf[goDf$p.adjust &lt; 0.05, c(1:5,7)]</p>
<div id="section-6" class="section level48">
<p class="heading"><span class="header-section-number">2.2.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1</span> </p>
<p>#make heatmap of differentially expressed genes#
################################################</p>
<p>exprs_heatmap = vGene$E[rownames(sigGene),]</p>
<p>df &lt;- as.data.frame(colData(rse_gene)[,c(“PrimaryDx”)])
rownames(df) &lt;- colnames(exprs_heatmap)
colnames(df)&lt;-“diagnosis”</p>
<p>pdf(file=here(“DE_analysis”, “pdfs”, “de_heatmap.pdf”))
pheatmap(exprs_heatmap, cluster_rows=TRUE, show_rownames=FALSE,
cluster_cols=TRUE, annotation_col=df)
dev.off()</p>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": {
      styles: {
        ".MathJax_Display": {
           "text-align": "center",
           padding: "0px 150px 0px 65px",
           margin: "0px 0px 0.5em"
        },
        "@media screen and (max-width: 991px)": {
            ".MathJax_Display": {
               "text-align": "center",
               padding: "0 0 0 0"
            }
         }
      }
    }
  });
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
