<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2020-11-19" />

<title>SPEAQeasy Example Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">

</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="/__w/_temp/Library/BiocStyle/resources/html/bioconductor.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 828px;
  margin-left: auto;
  margin-right: auto;
}

img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {

}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 246px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



<script>
function toggle_visibility(id1) {
  var e = document.getElementById(id1);
  e.style.display = ((e.style.display!="none") ? "none" : "block");
}
</script>

</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SPEAQeasy example</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="prepare_data.html">Download Example Data</a>
</li>
<li>
  <a href="run_speaqeasy.html">Run SPEAQeasy</a>
</li>
<li>
  <a href="swap_speaqeasy.html">Resolve Sample Identities</a>
</li>
<li>
  <a href="de_analysis_speaqeasy.html">Differential Expression</a>
</li>
<li>
  <a href="bootcamp_intro">Bootcamp</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/LieberInstitute/SPEAQeasy-example">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">SPEAQeasy Example Analysis</h1>
<p class="author-name">Nicholas J. Eagles<span class="affil-mark">1</span> and Leonardo Collado-Torres<span class="affil-mark">1,2*</span></p>
<p class="author-affiliation"><span class="affil-mark">1</span>Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><span class="affil-mark">2</span>Center for Computational Biology, Johns Hopkins University</p>
<p class="author-email"><span class="affil-mark">*</span><a href="mailto:lcolladotor@gmail.com">lcolladotor@gmail.com</a></p>
<h4 class="date">19 November 2020</h4>

</div>


<div id="analysis" class="section level1">
<h1><span class="header-section-number">1</span> Analysis</h1>
<p>A guided example showing how processed results from the RNAseq pipeline <a href="https://github.com/LieberInstitute/SPEAQeasy">SPEAQeasy</a> can be used in differential expression analyses and visualization. We also show how existing genotype information for a set of samples can be combined with SPEAQeasy results to resolve any identity issues, as can emerge during sequencing.</p>
<ul>
<li><a href="prepare_data.html">Downloading the example data</a></li>
<li><a href="run_speaqeasy.html">Run SPEAQeasy on your setup</a></li>
<li><a href="swap_speaqeasy.html">Resolve identity issues and attach phenotype data</a></li>
<li><a href="de_analysis_speaqeasy.html">Perform differential expression and visualize results</a></li>
</ul>
</div>
<div id="cite-speaqeasy" class="section level1">
<h1><span class="header-section-number">2</span> Cite <code>SPEAQeasy</code></h1>
<p>We hope that <a href="http://research.libd.org/SPEAQeasy/"><code>SPEAQeasy</code></a> will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<pre><code>ADD CITATION INFO</code></pre>
<p>This is a project by the <a href="https://lcolladotor.github.io/bioc_team_ds">R/Bioconductor-powered Team Data Science</a> at the <a href="https://www.libd.org/">Lieber Institute for Brain Development</a>.</p>
<div class='horizontal-scroll'><table>
<tbody>
<tr class="odd">
<td><a href="https://lcolladotor.github.io/bioc_team_ds"><img src="http://lcolladotor.github.io/img/Leo_transparente.png" width="250px"></a></td>
<td><a href="https://www.libd.org/"><img src="http://lcolladotor.github.io/img/LIBD_logo.jpg" width="250px"></a></td>
</tr>
</tbody>
</table></div>
</div>
<div id="references" class="section level1">
<h1><span class="header-section-number">3</span> References</h1>
<p>The analysis was possible thanks to R and many of its packages:</p>
<ul>
<li>R (<a href='https://www.R-project.org/'>R Core Team, 2020</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/BiocStyle">BiocStyle</a></em> (<a href='https://github.com/rstudio/rmarkdown'>Allaire, Xie, McPherson, Luraschi, et al., 2020</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/clusterProfiler">clusterProfiler</a></em> (<a href='#bib-clusterProfiler'>Carlson, 2020</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=devtools">devtools</a></em> (<a href='https://github.com/cboettig/knitcitations'>Boettiger, 2020</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/edgeR">edgeR</a></em> (<a href='http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html'>Collado-Torres, Nellore, Kammers, Ellis, et al., 2017</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=getopt">getopt</a></em> (<a href='https://github.com/LieberInstitute/jaffelab'>Collado-Torres, Jaffe, and Burke, 2019</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=here">here</a></em> (<a href='https://github.com/trevorld/r-getopt'>Davis and Day, 2019</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> (<a href='https://bioconductor.org/packages/SummarizedExperiment'>Morgan, Obenchain, Hester, and Pagès, 2020</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=jaffelab">jaffelab</a></em> (<a href='#bib-jaffelab'>Kolde, 2019</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> (<a href='#bib-knitcitations'>Müller, 2020</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/org.Hs.eg.db">org.Hs.eg.db</a></em> (<a href='#bib-orgHsegdb'>Neuwirth, 2014</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=pheatmap">pheatmap</a></em> (<a href='https://doi.org/10.1093/bioinformatics/btu168'>Obenchain, Lawrence, Carey, Gogarten, et al., 2014</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=RColorBrewer">RColorBrewer</a></em> (<a href='https://github.com/Bioconductor/BiocStyle'>Oleś, Morgan, and Huber, 2020</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/recount">recount</a></em> (<a href='https://doi.org/10.1093/nar/gkv007'>Ritchie, Phipson, Wu, Hu, et al., 2015</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> (<a href='https://doi.org/10.1093/bioinformatics/btp616'>Robinson, McCarthy, and Smyth, 2010</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em> (<a href='#bib-SummarizedExperiment'>Law, Chen, Shi, and Smyth, 2014</a>)</li>
<li><em><a href="https://CRAN.R-project.org/package=tidyr">tidyr</a></em> (<a href='#bib-tidyr'>Wickham, Hester, and Chang, 2020</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/VariantAnnotation">VariantAnnotation</a></em> (<a href='#bib-VariantAnnotation'>Wickham, 2020</a>)</li>
<li><em><a href="https://bioconductor.org/packages/3.12/voom">voom</a></em> (<a href='https://doi.org/10.1089/omi.2011.0118'>Yu, Wang, Han, and He, 2012</a>)</li>
</ul>
<p><a href="index.bib">Bibliography file</a></p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, J. Luraschi, et al. <em>rmarkdown: Dynamic Documents for R</em>. R package
version 2.5. 2020. &lt;URL: <a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for ‘Knitr’ Markdown Files</em>. R package version 1.0.11. 2020.
&lt;URL: <a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>&gt;.</p>
<p>[3] M. Carlson. <em>org.Hs.eg.db: Genome wide annotation for Human</em>. R package version 3.12.0. 2020.</p>
<p>[4] L. Collado-Torres, A. E. Jaffe, and E. E. Burke. <em>jaffelab: Commonly used functions by the Jaffe lab</em>.
R package version 0.99.30. 2019. &lt;URL: <a href="https://github.com/LieberInstitute/jaffelab" class="uri">https://github.com/LieberInstitute/jaffelab</a>&gt;.</p>
<p>[5] L. Collado-Torres, A. Nellore, K. Kammers, S. E. Ellis, et al. “Reproducible RNA-seq analysis using
recount2”. In: <em>Nature Biotechnology</em> (2017). DOI: 10.1038/nbt.3838. &lt;URL:
<a href="http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html" class="uri">http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html</a>&gt;.</p>
<p>[6] T. L. Davis and A. Day. <em>getopt: C-Like ‘getopt’ Behavior</em>. R package version 1.20.3. 2019. &lt;URL:
<a href="https://github.com/trevorld/r-getopt" class="uri">https://github.com/trevorld/r-getopt</a>&gt;.</p>
<p>[7] R. Kolde. <em>pheatmap: Pretty Heatmaps</em>. R package version 1.0.12. 2019.</p>
<p>[8] C. Law, Y. Chen, W. Shi, and G. Smyth. “Voom: precision weights unlock linear model analysis tools for
RNA-seq read counts”. In: <em>Genome Biology</em> 15 (2014), p. R29.</p>
<p>[9] M. Morgan, V. Obenchain, J. Hester, and H. Pagès. <em>SummarizedExperiment: SummarizedExperiment
container</em>. R package version 1.20.0. 2020. &lt;URL: <a href="https://bioconductor.org/packages/SummarizedExperiment" class="uri">https://bioconductor.org/packages/SummarizedExperiment</a>&gt;.</p>
<p>[10] K. Müller. <em>here: A Simpler Way to Find Your Files</em>. <a href="https://here.r-lib.org/" class="uri">https://here.r-lib.org/</a>,
<a href="https://github.com/r-lib/here" class="uri">https://github.com/r-lib/here</a>. 2020.</p>
<p>[11] E. Neuwirth. <em>RColorBrewer: ColorBrewer Palettes</em>. R package version 1.1-2. 2014.</p>
<p>[12] V. Obenchain, M. Lawrence, V. Carey, S. Gogarten, et al. “VariantAnnotation: a Bioconductor package
for exploration and annotation of genetic variants”. In: <em>Bioinformatics</em> 30.14 (2014), pp. 2076-2078. DOI:
10.1093/bioinformatics/btu168.</p>
<p>[13] A. Oleś, M. Morgan, and W. Huber. <em>BiocStyle: Standard styles for vignettes and other Bioconductor
documents</em>. R package version 2.18.0. 2020. &lt;URL: <a href="https://github.com/Bioconductor/BiocStyle" class="uri">https://github.com/Bioconductor/BiocStyle</a>&gt;.</p>
<p>[14] R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical
Computing. Vienna, Austria, 2020. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[15] M. E. Ritchie, B. Phipson, D. Wu, Y. Hu, et al. “limma powers differential expression analyses for
RNA-sequencing and microarray studies”. In: <em>Nucleic Acids Research</em> 43.7 (2015), p. e47. DOI:
10.1093/nar/gkv007.</p>
<p>[16] M. D. Robinson, D. J. McCarthy, and G. K. Smyth. “edgeR: a Bioconductor package for differential
expression analysis of digital gene expression data”. In: <em>Bioinformatics</em> 26.1 (2010), pp. 139-140. DOI:
10.1093/bioinformatics/btp616.</p>
<p>[17] H. Wickham. <em>tidyr: Tidy Messy Data</em>. <a href="https://tidyr.tidyverse.org" class="uri">https://tidyr.tidyverse.org</a>, <a href="https://github.com/tidyverse/tidyr" class="uri">https://github.com/tidyverse/tidyr</a>.
2020.</p>
<p>[18] H. Wickham, J. Hester, and W. Chang. <em>devtools: Tools to Make Developing R Packages Easier</em>.
<a href="https://devtools.r-lib.org/" class="uri">https://devtools.r-lib.org/</a>, <a href="https://github.com/r-lib/devtools" class="uri">https://github.com/r-lib/devtools</a>. 2020.</p>
<p>[19] G. Yu, L. Wang, Y. Han, and Q. He. “clusterProfiler: an R package for comparing biological themes
among gene clusters”. In: <em>OMICS: A Journal of Integrative Biology</em> 16.5 (2012), pp. 284-287. DOI:
10.1089/omi.2011.0118.</p>
<pre class="r"><code># Time spent creating this report:
diff(c(timestart, Sys.time()))</code></pre>
<pre><code>## Time difference of 2.827932 mins</code></pre>
<pre class="r"><code># Date this report was generated
message(Sys.time())</code></pre>
<pre><code>## 2020-11-19 17:56:13</code></pre>
<pre class="r"><code># Reproducibility info
options(width = 120)
sessioninfo::session_info()</code></pre>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 4.0.3 (2020-10-10)
##  os       Ubuntu 20.04 LTS            
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       Etc/UTC                     
##  date     2020-11-19                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version  date       lib source                                   
##  AnnotationDbi        * 1.52.0   2020-10-27 [1] Bioconductor                             
##  askpass                1.1      2019-01-13 [2] RSPM (R 4.0.0)                           
##  assertthat             0.2.1    2019-03-21 [2] RSPM (R 4.0.0)                           
##  backports              1.2.0    2020-11-02 [2] RSPM (R 4.0.3)                           
##  base64enc              0.1-3    2015-07-28 [2] RSPM (R 4.0.0)                           
##  bibtex                 0.4.3    2020-11-18 [1] Github (ropensci/bibtex@7ccf8ea)         
##  Biobase              * 2.50.0   2020-10-27 [1] Bioconductor                             
##  BiocFileCache          1.14.0   2020-10-27 [1] Bioconductor                             
##  BiocGenerics         * 0.36.0   2020-10-27 [1] Bioconductor                             
##  BiocManager          * 1.30.10  2019-11-16 [2] CRAN (R 4.0.3)                           
##  BiocParallel           1.24.1   2020-11-06 [1] Bioconductor                             
##  BiocStyle            * 2.18.0   2020-10-27 [1] Bioconductor                             
##  biomaRt                2.46.0   2020-10-27 [1] Bioconductor                             
##  Biostrings             2.58.0   2020-10-27 [1] Bioconductor                             
##  bit                    4.0.4    2020-08-04 [1] RSPM (R 4.0.2)                           
##  bit64                  4.0.5    2020-08-30 [1] RSPM (R 4.0.2)                           
##  bitops                 1.0-6    2013-08-17 [1] RSPM (R 4.0.0)                           
##  blob                   1.2.1    2020-01-20 [1] RSPM (R 4.0.0)                           
##  bookdown               0.21     2020-10-13 [1] RSPM (R 4.0.2)                           
##  BSgenome               1.58.0   2020-10-27 [1] Bioconductor                             
##  bumphunter             1.32.0   2020-10-27 [1] Bioconductor                             
##  Cairo                  1.5-12.2 2020-07-07 [1] RSPM (R 4.0.3)                           
##  callr                  3.5.1    2020-10-13 [2] RSPM (R 4.0.2)                           
##  checkmate              2.0.0    2020-02-06 [1] RSPM (R 4.0.0)                           
##  cli                    2.1.0    2020-10-12 [2] RSPM (R 4.0.2)                           
##  cluster                2.1.0    2019-06-19 [3] CRAN (R 4.0.3)                           
##  clusterProfiler      * 3.18.0   2020-10-27 [1] Bioconductor                             
##  codetools              0.2-18   2020-11-04 [3] RSPM (R 4.0.3)                           
##  colorspace             2.0-0    2020-11-11 [1] RSPM (R 4.0.3)                           
##  cowplot                1.1.0    2020-09-08 [1] RSPM (R 4.0.2)                           
##  crayon                 1.3.4    2017-09-16 [2] RSPM (R 4.0.0)                           
##  crosstalk              1.1.0.1  2020-03-13 [2] RSPM (R 4.0.0)                           
##  curl                   4.3      2019-12-02 [2] RSPM (R 4.0.0)                           
##  data.table             1.13.2   2020-10-19 [1] RSPM (R 4.0.3)                           
##  DBI                    1.1.0    2019-12-15 [1] RSPM (R 4.0.0)                           
##  dbplyr                 2.0.0    2020-11-03 [1] RSPM (R 4.0.3)                           
##  DelayedArray           0.16.0   2020-10-27 [1] Bioconductor                             
##  derfinder              1.24.0   2020-10-27 [1] Bioconductor                             
##  derfinderHelper        1.24.0   2020-10-27 [1] Bioconductor                             
##  desc                   1.2.0    2018-05-01 [2] RSPM (R 4.0.0)                           
##  devtools             * 2.3.2    2020-09-18 [2] RSPM (R 4.0.2)                           
##  digest                 0.6.27   2020-10-24 [2] RSPM (R 4.0.3)                           
##  DO.db                  2.9      2020-11-18 [1] Bioconductor                             
##  doRNG                  1.8.2    2020-01-27 [1] RSPM (R 4.0.0)                           
##  DOSE                   3.16.0   2020-10-27 [1] Bioconductor                             
##  downloader             0.4      2015-07-09 [1] RSPM (R 4.0.0)                           
##  dplyr                  1.0.2    2020-08-18 [1] RSPM (R 4.0.2)                           
##  DT                     0.16     2020-10-13 [2] RSPM (R 4.0.2)                           
##  edgeR                * 3.32.0   2020-10-27 [1] Bioconductor                             
##  ellipsis               0.3.1    2020-05-15 [2] RSPM (R 4.0.0)                           
##  enrichplot           * 1.10.1   2020-11-14 [1] Bioconductor                             
##  evaluate               0.14     2019-05-28 [2] RSPM (R 4.0.0)                           
##  ExploreModelMatrix   * 1.2.0    2020-10-27 [1] Bioconductor                             
##  fansi                  0.4.1    2020-01-08 [2] RSPM (R 4.0.0)                           
##  farver                 2.0.3    2020-01-16 [1] RSPM (R 4.0.0)                           
##  fastmap                1.0.1    2019-10-08 [1] RSPM (R 4.0.0)                           
##  fastmatch              1.1-0    2017-01-28 [1] RSPM (R 4.0.0)                           
##  fgsea                  1.16.0   2020-10-27 [1] Bioconductor                             
##  foreach                1.5.1    2020-10-15 [1] RSPM (R 4.0.2)                           
##  foreign                0.8-80   2020-05-24 [3] CRAN (R 4.0.3)                           
##  Formula                1.2-4    2020-10-16 [1] RSPM (R 4.0.2)                           
##  fs                     1.5.0    2020-07-31 [2] RSPM (R 4.0.2)                           
##  generics               0.1.0    2020-10-31 [1] RSPM (R 4.0.3)                           
##  GenomeInfoDb         * 1.26.0   2020-10-27 [1] Bioconductor                             
##  GenomeInfoDbData       1.2.4    2020-11-18 [1] Bioconductor                             
##  GenomicAlignments      1.26.0   2020-10-27 [1] Bioconductor                             
##  GenomicFeatures        1.42.1   2020-11-12 [1] Bioconductor                             
##  GenomicFiles           1.26.0   2020-10-27 [1] Bioconductor                             
##  GenomicRanges        * 1.42.0   2020-10-27 [1] Bioconductor                             
##  GEOquery               2.58.0   2020-10-27 [1] Bioconductor                             
##  ggforce                0.3.2    2020-06-23 [1] RSPM (R 4.0.2)                           
##  ggnewscale             0.4.3    2020-08-27 [1] RSPM (R 4.0.2)                           
##  ggplot2              * 3.3.2    2020-06-19 [1] RSPM (R 4.0.1)                           
##  ggraph                 2.0.4    2020-11-16 [1] RSPM (R 4.0.3)                           
##  ggrepel                0.8.2    2020-03-08 [1] RSPM (R 4.0.2)                           
##  glue                   1.4.2    2020-08-27 [2] RSPM (R 4.0.2)                           
##  GO.db                  3.12.1   2020-11-18 [1] Bioconductor                             
##  googledrive            1.0.1    2020-05-05 [1] RSPM (R 4.0.0)                           
##  GOSemSim               2.16.1   2020-10-29 [1] Bioconductor                             
##  graphlayouts           0.7.1    2020-10-26 [1] RSPM (R 4.0.3)                           
##  gridExtra              2.3      2017-09-09 [1] RSPM (R 4.0.0)                           
##  gtable                 0.3.0    2019-03-25 [1] RSPM (R 4.0.0)                           
##  here                 * 1.0.0    2020-11-15 [1] RSPM (R 4.0.3)                           
##  Hmisc                  4.4-1    2020-08-10 [1] RSPM (R 4.0.2)                           
##  hms                    0.5.3    2020-01-08 [1] RSPM (R 4.0.0)                           
##  htmlTable              2.1.0    2020-09-16 [1] RSPM (R 4.0.2)                           
##  htmltools              0.5.0    2020-06-16 [2] RSPM (R 4.0.1)                           
##  htmlwidgets            1.5.2    2020-10-03 [2] RSPM (R 4.0.2)                           
##  httpuv                 1.5.4    2020-06-06 [1] RSPM (R 4.0.2)                           
##  httr                   1.4.2    2020-07-20 [2] RSPM (R 4.0.2)                           
##  igraph                 1.2.6    2020-10-06 [1] RSPM (R 4.0.2)                           
##  IRanges              * 2.24.0   2020-10-27 [1] Bioconductor                             
##  iterators              1.0.13   2020-10-15 [1] RSPM (R 4.0.2)                           
##  jaffelab             * 0.99.30  2020-11-18 [1] Github (LieberInstitute/jaffelab@42637ff)
##  jpeg                   0.1-8.1  2019-10-24 [1] RSPM (R 4.0.0)                           
##  jsonlite               1.7.1    2020-09-07 [2] RSPM (R 4.0.2)                           
##  knitcitations        * 1.0.11   2020-11-18 [1] Github (cboettig/knitcitations@23750d5)  
##  knitr                  1.30     2020-09-22 [2] RSPM (R 4.0.2)                           
##  labeling               0.4.2    2020-10-20 [1] RSPM (R 4.0.3)                           
##  later                  1.1.0.1  2020-06-05 [2] RSPM (R 4.0.2)                           
##  lattice                0.20-41  2020-04-02 [3] CRAN (R 4.0.3)                           
##  latticeExtra           0.6-29   2019-12-19 [1] RSPM (R 4.0.0)                           
##  lazyeval               0.2.2    2019-03-15 [2] RSPM (R 4.0.0)                           
##  lifecycle              0.2.0    2020-03-06 [2] RSPM (R 4.0.0)                           
##  limma                * 3.46.0   2020-10-27 [1] Bioconductor                             
##  locfit                 1.5-9.4  2020-03-25 [1] RSPM (R 4.0.0)                           
##  lubridate              1.7.9.2  2020-11-13 [1] RSPM (R 4.0.3)                           
##  magrittr               2.0.1    2020-11-17 [2] RSPM (R 4.0.3)                           
##  MASS                   7.3-53   2020-09-09 [3] CRAN (R 4.0.3)                           
##  Matrix                 1.2-18   2019-11-27 [3] CRAN (R 4.0.3)                           
##  MatrixGenerics       * 1.2.0    2020-10-27 [1] Bioconductor                             
##  matrixStats          * 0.57.0   2020-09-25 [1] RSPM (R 4.0.2)                           
##  memoise                1.1.0    2017-04-21 [2] RSPM (R 4.0.0)                           
##  mime                   0.9      2020-02-04 [2] RSPM (R 4.0.0)                           
##  munsell                0.5.0    2018-06-12 [1] RSPM (R 4.0.0)                           
##  nnet                   7.3-14   2020-04-26 [3] CRAN (R 4.0.3)                           
##  openssl                1.4.3    2020-09-18 [2] RSPM (R 4.0.2)                           
##  org.Hs.eg.db         * 3.12.0   2020-11-19 [1] Bioconductor                             
##  pheatmap             * 1.0.12   2019-01-04 [1] RSPM (R 4.0.0)                           
##  pillar                 1.4.6    2020-07-10 [2] RSPM (R 4.0.2)                           
##  pkgbuild               1.1.0    2020-07-13 [2] RSPM (R 4.0.2)                           
##  pkgconfig              2.0.3    2019-09-22 [2] RSPM (R 4.0.0)                           
##  pkgload                1.1.0    2020-05-29 [2] RSPM (R 4.0.0)                           
##  plotly               * 4.9.2.1  2020-04-04 [1] RSPM (R 4.0.0)                           
##  plyr                   1.8.6    2020-03-03 [1] RSPM (R 4.0.2)                           
##  png                    0.1-7    2013-12-03 [1] RSPM (R 4.0.0)                           
##  polyclip               1.10-0   2019-03-14 [1] RSPM (R 4.0.0)                           
##  prettyunits            1.1.1    2020-01-24 [2] RSPM (R 4.0.0)                           
##  processx               3.4.4    2020-09-03 [2] RSPM (R 4.0.2)                           
##  progress               1.2.2    2019-05-16 [1] RSPM (R 4.0.0)                           
##  promises               1.1.1    2020-06-09 [2] RSPM (R 4.0.2)                           
##  ps                     1.4.0    2020-10-07 [2] RSPM (R 4.0.2)                           
##  purrr                  0.3.4    2020-04-17 [2] RSPM (R 4.0.0)                           
##  qvalue                 2.22.0   2020-10-27 [1] Bioconductor                             
##  R6                     2.5.0    2020-10-28 [2] RSPM (R 4.0.3)                           
##  rafalib              * 1.0.0    2015-08-09 [1] RSPM (R 4.0.0)                           
##  rappdirs               0.3.1    2016-03-28 [1] RSPM (R 4.0.0)                           
##  RColorBrewer         * 1.1-2    2014-12-07 [1] RSPM (R 4.0.0)                           
##  Rcpp                   1.0.5    2020-07-06 [2] RSPM (R 4.0.2)                           
##  RCurl                  1.98-1.2 2020-04-18 [1] RSPM (R 4.0.0)                           
##  readr                  1.4.0    2020-10-05 [1] RSPM (R 4.0.3)                           
##  recount              * 1.16.0   2020-10-27 [1] Bioconductor                             
##  RefManageR             1.3.0    2020-11-18 [1] Github (ropensci/RefManageR@ab8fe60)     
##  remotes                2.2.0    2020-07-21 [2] RSPM (R 4.0.2)                           
##  rentrez                1.2.3    2020-11-10 [1] RSPM (R 4.0.3)                           
##  reshape2               1.4.4    2020-04-09 [1] RSPM (R 4.0.2)                           
##  rintrojs               0.2.2    2019-05-29 [1] RSPM (R 4.0.0)                           
##  rlang                  0.4.8    2020-10-08 [2] RSPM (R 4.0.2)                           
##  rmarkdown              2.5      2020-10-21 [1] RSPM (R 4.0.3)                           
##  rngtools               1.5      2020-01-23 [1] RSPM (R 4.0.0)                           
##  rpart                  4.1-15   2019-04-12 [3] CRAN (R 4.0.3)                           
##  rprojroot              2.0.2    2020-11-15 [2] RSPM (R 4.0.3)                           
##  Rsamtools              2.6.0    2020-10-27 [1] Bioconductor                             
##  RSQLite                2.2.1    2020-09-30 [1] RSPM (R 4.0.2)                           
##  rstudioapi             0.13     2020-11-12 [2] RSPM (R 4.0.3)                           
##  rtracklayer            1.50.0   2020-10-27 [1] Bioconductor                             
##  rvcheck                0.1.8    2020-03-01 [1] RSPM (R 4.0.0)                           
##  S4Vectors            * 0.28.0   2020-10-27 [1] Bioconductor                             
##  scales                 1.1.1    2020-05-11 [1] RSPM (R 4.0.0)                           
##  scatterpie             0.1.5    2020-09-09 [1] RSPM (R 4.0.2)                           
##  segmented              1.3-0    2020-10-27 [1] RSPM (R 4.0.3)                           
##  sessioninfo            1.1.1    2018-11-05 [1] RSPM (R 4.0.0)                           
##  shadowtext             0.0.7    2019-11-06 [1] RSPM (R 4.0.0)                           
##  shiny                  1.5.0    2020-06-23 [1] RSPM (R 4.0.2)                           
##  shinydashboard         0.7.1    2018-10-17 [1] RSPM (R 4.0.0)                           
##  shinyjs                2.0.0    2020-09-09 [1] RSPM (R 4.0.2)                           
##  statmod                1.4.35   2020-10-19 [1] RSPM (R 4.0.3)                           
##  stringi                1.5.3    2020-09-09 [2] RSPM (R 4.0.2)                           
##  stringr                1.4.0    2019-02-10 [2] RSPM (R 4.0.0)                           
##  SummarizedExperiment * 1.20.0   2020-10-27 [1] Bioconductor                             
##  survival               3.2-7    2020-09-28 [3] CRAN (R 4.0.3)                           
##  testthat               3.0.0    2020-10-31 [2] RSPM (R 4.0.3)                           
##  tibble                 3.0.4    2020-10-12 [2] RSPM (R 4.0.2)                           
##  tidygraph              1.2.0    2020-05-12 [1] RSPM (R 4.0.2)                           
##  tidyr                  1.1.2    2020-08-27 [1] RSPM (R 4.0.3)                           
##  tidyselect             1.1.0    2020-05-11 [1] RSPM (R 4.0.0)                           
##  tweenr                 1.0.1    2018-12-14 [1] RSPM (R 4.0.2)                           
##  usethis              * 1.6.3    2020-09-17 [1] RSPM (R 4.0.2)                           
##  VariantAnnotation      1.36.0   2020-10-27 [1] Bioconductor                             
##  vctrs                  0.3.5    2020-11-17 [2] RSPM (R 4.0.3)                           
##  viridis                0.5.1    2018-03-29 [1] RSPM (R 4.0.0)                           
##  viridisLite            0.3.0    2018-02-01 [1] RSPM (R 4.0.0)                           
##  withr                  2.3.0    2020-09-22 [2] RSPM (R 4.0.2)                           
##  xfun                   0.19     2020-10-30 [2] RSPM (R 4.0.3)                           
##  XML                    3.99-0.5 2020-07-23 [1] RSPM (R 4.0.2)                           
##  xml2                   1.3.2    2020-04-23 [2] RSPM (R 4.0.0)                           
##  xtable                 1.8-4    2019-04-21 [1] RSPM (R 4.0.0)                           
##  XVector                0.30.0   2020-10-27 [1] Bioconductor                             
##  yaml                   2.2.1    2020-02-01 [2] RSPM (R 4.0.0)                           
##  zlibbioc               1.36.0   2020-10-27 [1] Bioconductor                             
## 
## [1] /__w/_temp/Library
## [2] /usr/local/lib/R/site-library
## [3] /usr/local/lib/R/library</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": {
      styles: {
        ".MathJax_Display": {
           "text-align": "center",
           padding: "0px 150px 0px 65px",
           margin: "0px 0px 0.5em"
        },
        "@media screen and (max-width: 991px)": {
            ".MathJax_Display": {
               "text-align": "center",
               padding: "0 0 0 0"
            }
         }
      }
    }
  });
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
